Mjolnir/src/OI.h

```diff
@@ -35,11 +35,48 @@ class OI {
 public:
 	OI();
 
-	// BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=PROTOTYPES
-
-	std::shared_ptr<Joystick> getdriver();
-	std::shared_ptr<Joystick> getcoDriver();
+	// BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=PROTOTYP
 	// END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=PROTOTYPES
+	static std::shared_ptr<OI> self;
+	std::shared_ptr<Joystick> driver;
+	JoystickButton dA;
+	JoystickButton dB;
+	JoystickButton dX;
+	JoystickButton dY;
+	JoystickButton dLB;
+	JoystickButton dRB;
+	std::shared_ptr<Joystick> coDriver;
+	JoystickButton coA;
+	JoystickButton coB;
+	JoystickButton coX;
+	JoystickButton coY;
+	JoystickButton coLB;
+	JoystickButton coRB;
+public:
+	static std::shared_ptr<OI> getInstance();
+	OI();
+	std::shared_ptr<Joystick> getDriver();
+	double getDriverLeftXAxis() const;
+	double getDriverLeftYAxis() const;
+	double getDriverRightXAxis() const;
+	double getDriverRightYAxis() const;
+	std::shared_ptr<Joystick> getCoDriver();
+	double getCoDriverLeftXAxis() const;
+	double getCoDriverLeftYAxis() const;
+	double getCoDriverRightXAxis() const;
+	double getCoDriverRightYAxis() const;
+
+	static const int kButtonA_ID = 1;
```

Comment:
These `static` make sense. To clarify, no static variables. These are proper constants.